src/sinks/vector/config.rs

```diff
@@ -253,6 +253,8 @@ impl RetryLogic for VectorGrpcRetryLogic {
                     | OutOfRange
                     | Unimplemented
                     | Unauthenticated
+                    | Internal
+                    | DataLoss
```

Comment:
`Internal` errors occur when `At least one event in the batch had a transient error in delivery` and `DataLoss` errors occur when `At least one event in the batch had a permanent failure or rejection`, with `Internal` taking precedence over `Dataloss`.

https://github.com/vectordotdev/vector/blob/a141d2b163aa487a15c278096c24a57849b53d8a/lib/vector-common/src/finalization.rs#L283

With that in mind, we'll want to retry for `Internal`, but not `DataLoss`.

```suggestion
                    | DataLoss
```